server/src/main/java/org/opensearch/cluster/metadata/WeightedRoutingMetadata.java

```diff
@@ -76,6 +87,7 @@ public Version getMinimalSupportedVersion() {
     public void writeTo(StreamOutput out) throws IOException {
         if (weightedRouting != null) {
             weightedRouting.writeTo(out);
+            out.writeLong(version);
```

Comment:
Is it fine to break compatibility if the API stability is marked `experimental`. Alternatively we could add a OpenSearch version check for newer attributes

Reply:
OpenSearch rest api spec says
```
experimental highly likely to break in the near future (minor/path), no bwc guarantees. Possibly removed in the future.
```
https://github.com/opensearch-project/OpenSearch/blob/main/rest-api-spec/README.md
So I think we can break compatibility. What are your thoughts on this?